metrics_sdk/lib/opentelemetry/sdk/metrics/state/asynchronous_metric_stream.rb

```diff
@@ -48,28 +50,53 @@ def collect(start_time, end_time)
           def invoke_callback(timeout, attributes)
             if @registered_views.empty?
               @mutex.synchronize do
-                Timeout.timeout(timeout || 30) do
-                  @callback.each do |cb|
-                    value = cb.call
-                    @default_aggregation.update(value, attributes, @data_points)
-                  end
+                @callback.each do |cb|
```

Comment:
Since this PR includes changes to the timeout mechanism, I think the conventional commit type in the PR title should be `feat` or `fix` to initiate a release and add a changelog trail for users.